homeassistant/components/command_line/__init__.py

```diff
@@ -1 +1,177 @@
+        vol.Optional(CONF_DEVICE_CLASS): BINARY_SENSOR_DEVICE_CLASSES_SCHEMA,
+        vol.Optional(CONF_VALUE_TEMPLATE): cv.template,
+        vol.Optional(CONF_COMMAND_TIMEOUT, default=DEFAULT_TIMEOUT): cv.positive_int,
+        vol.Optional(CONF_UNIQUE_ID): cv.string,
+    }
+)
+COVER_SCHEMA = vol.Schema(
+    {
+        vol.Optional(CONF_COMMAND_CLOSE, default="true"): cv.string,
+        vol.Optional(CONF_COMMAND_OPEN, default="true"): cv.string,
+        vol.Optional(CONF_COMMAND_STATE): cv.string,
+        vol.Optional(CONF_COMMAND_STOP, default="true"): cv.string,
+        vol.Required(CONF_NAME): cv.string,
+        vol.Optional(CONF_VALUE_TEMPLATE): cv.template,
+        vol.Optional(CONF_COMMAND_TIMEOUT, default=DEFAULT_TIMEOUT): cv.positive_int,
+        vol.Optional(CONF_UNIQUE_ID): cv.string,
+    }
+)
+NOTIFY_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_COMMAND): cv.string,
+        vol.Optional(CONF_NAME): cv.string,
+        vol.Optional(CONF_COMMAND_TIMEOUT, default=DEFAULT_TIMEOUT): cv.positive_int,
+    }
+)
+SENSOR_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_COMMAND): cv.string,
+        vol.Optional(CONF_COMMAND_TIMEOUT, default=DEFAULT_TIMEOUT): cv.positive_int,
+        vol.Optional(CONF_JSON_ATTRIBUTES): cv.ensure_list_csv,
+        vol.Optional(CONF_NAME, default=SENSOR_DEFAULT_NAME): cv.string,
+        vol.Optional(CONF_UNIT_OF_MEASUREMENT): cv.string,
+        vol.Optional(CONF_VALUE_TEMPLATE): cv.template,
+        vol.Optional(CONF_UNIQUE_ID): cv.string,
+        vol.Optional(CONF_DEVICE_CLASS): SENSOR_DEVICE_CLASSES_SCHEMA,
+        vol.Optional(CONF_STATE_CLASS): SENSOR_STATE_CLASSES_SCHEMA,
+    }
+)
+SWITCH_SCHEMA = vol.Schema(
+    {
+        vol.Optional(CONF_COMMAND_OFF, default="true"): cv.string,
+        vol.Optional(CONF_COMMAND_ON, default="true"): cv.string,
+        vol.Optional(CONF_COMMAND_STATE): cv.string,
+        vol.Required(CONF_NAME): cv.string,
+        vol.Optional(CONF_VALUE_TEMPLATE): cv.template,
+        vol.Optional(CONF_ICON): cv.template,
+        vol.Optional(CONF_COMMAND_TIMEOUT, default=DEFAULT_TIMEOUT): cv.positive_int,
+        vol.Optional(CONF_UNIQUE_ID): cv.string,
+    }
+)
+COMBINED_SCHEMA = vol.Schema(
+    {
+        vol.Optional(BINARY_SENSOR_DOMAIN): BINARY_SENSOR_SCHEMA,
+        vol.Optional(COVER_DOMAIN): COVER_SCHEMA,
+        vol.Optional(NOTIFY_DOMAIN): NOTIFY_SCHEMA,
+        vol.Optional(SENSOR_DOMAIN): SENSOR_SCHEMA,
+        vol.Optional(SWITCH_DOMAIN): SWITCH_SCHEMA,
+    }
+)
+CONFIG_SCHEMA = vol.Schema(
+    {
+        vol.Optional(DOMAIN): vol.All(
+            cv.ensure_list,
+            [COMBINED_SCHEMA],
+        )
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
+    """Set up Command Line from yaml config."""
+    command_line_config: list[dict[str, dict[str, Any]]] = config.get(DOMAIN, {})
```

Comment:
The default should be an empty list as the domain config will be a list. We've even typed it as such.